dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgPanel.vue

```diff
@@ -113,6 +113,7 @@ export default defineComponent({
         kpiKey: kpiKey,
         kpiLabel: kpiField?.label ? kpiField.label : kpiKey,
         kpiDescription: kpiField?.description ? kpiField.description : "",
+        kpiFormFieldComponent: kpiField?.component ?? "",
         [dataIdOfLksgDataset]: kpiValue,
```

Comment:
Until now, we have used the terminology/hierarchy:
`lksgDataSet => areas => subareas => kpis`

Now you have introduced the types `Category` and `Subcategory`.

Therefore I suggest we also use this terminology consistently, e.g. in the `convertLksgDataToFrontendFormat()`.
The loops should loop through categories and subcategories.
Maybe you can even type the respective objects in the for loops with the correct type, e.g. the current `subAreaObject` could be of type `Subcategory` ?

Reply:
changed

Reply:
Discussed with Siegfried and adjusted.